src/realm.js

```diff
@@ -175,6 +175,7 @@ export class Realm {
     this.errorHandler = opts.errorHandler;
 
     this.globalSymbolRegistry = [];
+    this._abstractValuesDefined = []; // A list of nameStrings to ensure abstract values have unique names
```

Comment:
This will result in O(n) lookup perf. Use a `Set<string>` instead.

Reply:
While I understand keyed collections like Sets are supposed to use sublinear access times, Set still does a linear lookup of entries: [spec section](http://www.ecma-international.org/ecma-262/6.0/#sec-set.prototype.has). Am I misinterpreting it?

I had initially used a simple object as a lookup table to do better than O(n).

Reply:
It is possible to implement Sets to have O(1) lookup and any decent implementation will do so.

Reply:
Okay. I'll use the Sets approach immediately once all tests pass. On my TODO.